### PR TITLE
Add default fallback log group and stream names

### PR DESCRIPTION
**Pull request by davidnewhall**
*Issue #, if available:* #99

*Description of changes:*
This adds two new input variables to the plugin with defaulted static values. These variables represent fallback group and stream names that get used any time a `$(variable)` in one of the templated inputs fails to parse.

The build failed with these errors:
```
cloudwatch/helpers.go:75: Errorf format %w has unknown verb w
cloudwatch/helpers.go:98: Errorf format %w has unknown verb w
cloudwatch/helpers.go:116: Errorf format %w has unknown verb w
```
**The `%w` verb requires go 1.13 or newer.** Requires this: https://github.com/aws/aws-for-fluent-bit/pull/87

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by PettitWesley** [comment]
We can update the go version.

We should first update it here and check the build: https://github.com/aws/aws-for-fluent-bit/blob/master/Dockerfile.plugins#L1

And then update the build.yml for all the plugins: https://github.com/aws/amazon-cloudwatch-logs-for-fluent-bit/blob/mainline/.github/workflows/build.yml#L19

**Comment by davidnewhall** [comment]
very strange.... 
```
verifying github.com/aws/amazon-kinesis-firehose-for-fluent-bit@v1.4.1/go.mod: checksum mismatch
	downloaded: h1:LV5vSM9R3vODyhZM9jdUWLThms4Chcio46SUff6XfBY=
	go.sum:     h1:2hixxBh6Xygvxe6x/PSUTphh8dZ0WptVty98ftZIhAU=

SECURITY ERROR
This download does NOT match an earlier download recorded in go.sum.
The bits may have been replaced on the origin server, or an attacker may
have intercepted the download attempt.
```

I'm not sure what was causing the above error, but `amazon-kinesis-firehose-for-fluent-bit` had a newer version, so I updated `go.mod/sum` to this newer version and the error went away. I also [made a PR](https://github.com/aws/aws-for-fluent-bit/pull/87) for `aws-for-fluent-bit` to utilize go 1.15.

File: cloudwatch/cloudwatch.go [hunk 1]
```diff
@@ -440,19 +446,15 @@ func (output *OutputPlugin) setGroupStreamNames(e *Event) {
...
-	}
-
-	if e.group = s.buf.String(); len(e.group) == 0 {
-		e.group = output.logGroupName.String
-	}
-
-	if len(e.group) > maxGroupStreamLength {
+		e.group = output.defaultLogGroupName
+		logrus.Errorf("[cloudwatch %d] parsing log_group_name template '%s': %v",
+			output.PluginInstanceID, output.logGroupName.String, err)
```
Comment (PettitWesley) [inline cloudwatch/cloudwatch.go:451]: What is setting `err` here? 

Reading the code above this error log, I expected it to be something like "generated log group name is longer than max allowed value, using default group"

Comment (PettitWesley) [reply cloudwatch/cloudwatch.go:451]: Sorry I misread the code

File: cloudwatch/cloudwatch.go [hunk 2]
```diff
@@ -440,19 +446,15 @@ func (output *OutputPlugin) setGroupStreamNames(e *Event) {
...
-		logrus.Errorf("[cloudwatch %d] parsing log_group_name template: %v", output.PluginInstanceID, err)
-	}
-
-	if e.group = s.buf.String(); len(e.group) == 0 {
-		e.group = output.logGroupName.String
-	}
-
-	if len(e.group) > maxGroupStreamLength {
+		e.group = output.defaultLogGroupName
+		logrus.Errorf("[cloudwatch %d] parsing log_group_name template '%s': %v",
```
Comment (PettitWesley) [inline cloudwatch/cloudwatch.go]: may be add- "will use default group name" ?

**Review by PettitWesley — APPROVED** [review]
LGTM, one nit on an error string.

I am going to wait to merge this till after our planned release this Thursday for arm64 support, just to be safe.

We will need to update the other plugins to go 1.15, I'll add a note in our oncall log for that.

**Review by PettitWesley — APPROVED** [review]

**Pull request merged**